pkg/sac/context.go

```diff
@@ -37,3 +37,8 @@ func WithAllAccess(ctx context.Context) context.Context {
 func WithNoAccess(ctx context.Context) context.Context {
 	return WithGlobalAccessScopeChecker(ctx, denyAllScopeCheckerCore)
 }
+
+// CopyScopeCheckerCore from the source context to the destination context
+func CopyScopeCheckerCore(dstCtx, srcCtx context.Context) context.Context {
+	return context.WithValue(dstCtx, globalAccessScopeContextKey{}, srcCtx.Value(globalAccessScopeContextKey{}))
```

Comment:
```suggestion
	return WithGlobalAccessScopeChecker(dstCtx, GlobalAccessScopeChecker(srcCtx).Core())
```

Reply:
I actually prefer to not do this as someone could change the return value of Core and this function would not change (or if if GlobalAccessScopeChecker is nil then it would panic). I think the current code is a very clear copy